docs/User/ProjectRelated/Changelog.md

```diff
@@ -3,6 +3,13 @@
 <!-- See rendered changelog at https://streamly.composewell.com -->
 ## Unreleased
 
+* The unboxed arrays now use values of Prim over Storable. Please note/follow
+  the steps below for migration.
+  1. Make sure the values that you store in an unboxed array have the `Prim`
+     instance.
+  2. Use the type class constraint `Prim` instead of `Storable` in your
+     polymorphic code interfacing with unboxed arrays.
```

Comment:
This should be the only change suggested. The requirement of instance will be forced by the compiler, that can just be written as a note added to this.